File: master-package/bonus-webinars-2018/integration-tests-with-testcontainers/src/integration-test/java/com/testwithspring/master/OrderedDbUnitTestExecutionListener.java
package com.testwithspring.master;

import com.github.springtestdbunit.DbUnitTestExecutionListener;

/**
 * This test execution listener ensures that the {@link DbUnitTestExecutionListener}
 * is invoked before the {@link org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener}.
 * This ensures that we can insert the required user account data into our database
 * before the security context is created by the {@link org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener}.
 */
public class OrderedDbUnitTestExecutionListener extends DbUnitTestExecutionListener {

    @Override
    public int getOrder() {
        return 9999;
    }
}

File: master-package/bonus-lessons-2020/integration-tests-rest-api/src/integration-test/kotlin/com/testwithspring/master/OrderedDbUnitTestExecutionListener.kt
package com.testwithspring.master

import com.github.springtestdbunit.DbUnitTestExecutionListener

/**
 * This test execution listener ensures that the {@link DbUnitTestExecutionListener}
 * is invoked before the {@link org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener}.
 * This ensures that we can insert the required user account data into our database
 * before the security context is created by the {@link org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener}.
 */
class OrderedDbUnitTestExecutionListener: DbUnitTestExecutionListener() {

    override fun getOrder(): Int {
        return 9999
    }
}